### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by ChanceNCounter** [MycroftAI/lingua-franca#114]
As we handle multiple refactors, an increase in activity among community members (new languages!) and the addition of documentation, should we also create an informal roadmap?

I don't mean to constrain development, but rather to indicate clearly the module's known shortcomings and our intentions for addressing them.

I think we've collectively used the phrase "out of scope for this PR" enough to start writing it down.

It would also let us use some of the other GH features to help further categorize all these tickets, and track progress.

**Comment by rafaljanicki** [comment]
That's a great idea! If I'd be aware about the lang refactoring's timeline I probably stopped working until it's done, so yeah, that'd be quite helpful I believe :)

**Comment by ChanceNCounter** [comment]
> That's a great idea! If I'd be aware about the lang refactoring's timeline I probably stopped working until it's done, so yeah, that'd be quite helpful I believe :)

@rafaljanicki: I should have linked from the get-go, but you can absolutely keep working in the interim. We're looking at something like [this](https://github.com/ChanceNCounter/lingua-franca/blob/refactor/languages_cleanup/project-structure.md#on-localizing-functions). No need to worry about files that don't yet exist for you! 😛 

But, as long as you're working on your localized files in `lang`, and you do your best to make the function signatures match that description, there's no reason you can't just keep working. Your functions are black boxes, as far as the top level is concerned.

**Comment by ChanceNCounter** [comment]
After a few months and some thought, I've got a few ideas for possible milestones. These are in no particular order.

1) Time Function Overhaul: Some of the date- and time-related functions just need some improvement. Some are slated for a rewrite. Most need better localization, if only because localization tends to lag behind `en` (and occasionally `pt`.)

2) Full Localization: As of `v0.3.0`, LF supports and tracks what we call "full language codes," or the full BCP-47 code, as opposed to "primary language codes" (the first portion, 'en' or 'es', etc.) However, we don't really do anything with it. Few of the functions contain logic to properly utilize that information. We should set out to ensure that each and every occurrence of something that *can* be "fully localized" actually *is*.

3) Config: #128 could be a short-term milestone, and might help with localization stuff.

**Comment by krisgesling** [comment]
Is it worth starting a Project Board so we can have a visual roadmap of what's coming up and highlight what we think the priorities are?

Or would a collaborative document / wiki page be better?

**Comment by ChanceNCounter** [comment]
Personally, I like a board, though I'm not too wild about GitHub's GUI on that front. On the other hand, it's less likely to be seen by the community, and even less likely to be read. It looks too much like somebody's to-do list, gets hard to parse.

But if we're using the board with milestones, the combination will probably be pretty communicative, so there's that.
